x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStopDataFrameTransformAction.java

```diff
@@ -88,8 +91,31 @@ protected void doExecute(Task task, StopDataFrameTransformAction.Request request
                 new PageParams(0, 10_000),
                 request.isAllowNoMatch(),
                 ActionListener.wrap(hitsAndIds -> {
+                    if (request.isForce() == false) {
+                        PersistentTasksCustomMetaData tasks = state.metaData().custom(PersistentTasksCustomMetaData.TYPE);
+                        List<String> failedTasks = new ArrayList<>();
+                        List<String> failedReasons = new ArrayList<>();
+                        for (String transformId : hitsAndIds.v2()) {
+                            PersistentTasksCustomMetaData.PersistentTask<?> dfTask = tasks.getTask(transformId);
+                            if (dfTask.getState() instanceof DataFrameTransformState
+                                && ((DataFrameTransformState) dfTask.getState()).getTaskState() == DataFrameTransformTaskState.FAILED) {
+                                failedTasks.add(transformId);
+                                failedReasons.add(((DataFrameTransformState) dfTask.getState()).getReason());
+                            }
+                        }
+                        if (failedTasks.isEmpty() == false) {
+                            String msg = failedTasks.size() == 1 ?
+                                "Unable to stop data frame transform [" + request.getId()
```

Comment:
nit the same text is used in `taskOperation()`, could be extracted and refactored.